generator/php/build.gradle

```diff
@@ -48,39 +49,45 @@ task generateClient(type: GradleBuild) {
         
         def clientOutputDirPerVersion = clientOutputDir + "/" + parsedApiName;
         def specPath = it.path;
-        def packageBase =  criteoApiVersion;
-        if (packageBase.matches("[0-9].*"))
-             packageBase= "v" + packageBase;
+        def versionCategory = criteoApiVersion == 'preview' ? 'preview' : 'stable'
+  
+        def versionInNamespace =  criteoApiVersion
+        if (versionInNamespace.matches("[0-9].*"))
+             versionInNamespace= "v" + versionInNamespace
 
-        packageBase = "${criteoPackage}.${criteoService}.${packageBase}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
-        def packageVersion = "${criteoApiVersion}.${generatorVersion}.${getFormattedDate()}"; // example : 2021_10.1.211110
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
+        
+        def artifactVersion = versionCategory == 'preview' ? '0' : criteoApiVersion.replaceAll("_",".").toString()
+        artifactVersion += ".${generatorVersion}.${getFormattedDate()}".toString() // example : 2021.01.0.211110 for stable , 0.0.211110 for preview
 
-        def generateTask = task("openApiGenerate_php_" + parsedApiName, type: GenerateTask.class) {
+        def generateTask = task("openApiGenerate_${technologyStack}_" + parsedApiName, type: GenerateTask.class) {
                 configFile = "$projectDir/generatorConfiguration.yaml".toString()
-                generatorName = "php"
+                generatorName = "$language".toString()
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        invokerPackage          : "${packageBase}".toString(),
-                        packageUrl              : 'https://github.com/criteo/criteo-api-php-sdk',
+                        groupId                 : 'criteo',
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
```

Comment:
We decided not to put "preview" or "stable" in the package name anymore, so I think that it should just be "criteo-api-${criteoService}-sdk"

Reply:
Let's update the story accordingly, and take into account these latest decisions